### ISSUE TITLE
Directory is unavailable for user after upgrade to 14.0.1.

### ISSUE THREAD

**Issue by JoeatMJ** [nextcloud/server#11550]
### Steps to reproduce
1. Upgrade to 14.0.1 from 13.0.5 using command line
2. Login as a specific user
3.

### Expected behaviour
Page should show all folders and shares as normal.

### Actual behaviour
Error box at the top of the frame that says, "This directory is unavailable, please check the logs or contact the administrator"

### Server configuration
**Operating system**:
Ubuntu 16.04.1

**Web server:**
Apache/2.4.18 

**Database:**
mysql  Ver 14.14 Distrib 5.7.23, for Linux (x86_64)

**PHP version:**
v7.0.32-0ubuntu0.16.04.1

**Nextcloud version:** (see Nextcloud admin page)
14.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:**
13.0.5

**Where did you install Nextcloud from:**
Download direct from Nextcloud.
**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
No errors found.

```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - accessibility: 1.0.1
  - activity: 2.7.0
  - admin_audit: 1.4.0
  - announcementcenter: 3.3.0
  - audioplayer: 2.4.1
  - bruteforcesettings: 1.1.0
  - cloud_federation_api: 0.0.1
  - comments: 1.4.0
  - dav: 1.6.0
  - federatedfilesharing: 1.4.0
  - federation: 1.4.0
  - files: 1.9.0
  - files_pdfviewer: 1.3.2
  - files_retention: 1.3.0
  - files_sharing: 1.6.2
  - files_texteditor: 2.6.0
  - files_trashbin: 1.4.1
  - files_versions: 1.7.1
  - files_videoplayer: 1.3.0
  - firstrunwizard: 2.3.0
  - gallery: 18.1.0
  - groupfolders: 1.3.3
  - impersonate: 1.1.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.2.0
  - nextcloud_announcements: 1.3.0
  - notifications: 2.2.1
  - oauth2: 1.2.1
  - onlyoffice: 2.0.4
  - password_policy: 1.4.0
  - provisioning_api: 1.4.0
  - serverinfo: 1.4.0
  - sharebymail: 1.4.0
  - support: 1.0.0
  - survey_client: 1.2.0
  - systemtags: 1.4.0
  - theming: 1.5.0
  - twofactor_backupcodes: 1.3.1
  - updatenotification: 1.4.1
  - workflowengine: 1.4.0
Disabled:
  - encryption
  - files_external
  - user_external
  - user_ldap

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

 "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/nc.mjp.com",
        "dbtype": "mysql",
        "version": "14.0.1.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "log_rotate_size": 52428800,
        "onlyoffice": {
            "jwt_secret": "***REMOVED SENSITIVE VALUE***",
            "jwt_header": "***REMOVED SENSITIVE VALUE***"
        }
    }
}

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
NONE

**Are you using encryption:** yes/no
NO

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
NO

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
Firefox Edge, IE, Safari - All latest

**Operating system:**
Win10, MacOS 10.14

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)

<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
This is the only error I see in the log around the time of the user logging in:

Error
no app in context
OCP\Files\StorageNotAvailableException: File with id "85147" has not been found.
/var/www/html/lib/private/Files/Storage/Wrapper/Jail.php - line 218: 
OC\Files\Storage\FailedStorage->getPermissions("")
/var/www/html/apps/files_sharing/lib/SharedStorage.php - line 198: 
OC\Files\Storage\Wrapper\Jail->getPermissions("")
/var/www/html/lib/private/Files/Storage/Wrapper/Wrapper.php - line 214: 
OCA\Files_Sharing\SharedStorage->getPermissions("")
/var/www/html/apps/files_sharing/lib/Cache.php - line 145: 
OC\Files\Storage\Wrapper\Wrapper->getPermissions("")
/var/www/html/apps/files_sharing/lib/Cache.php - line 113: 
OCA\Files_Sharing\Cache->formatCacheEntry(OC\Files\Cache\CacheEntry {}, "")
/var/www/html/lib/private/Files/Node/Folder.php - line 307: 
OCA\Files_Sharing\Cache->get(85147)

OC\Files\Node\Folder->OC\Files\Node\{closure}("*** sensiti ... *")
/var/www/html/lib/private/Files/Node/Folder.php - line 321: 
array_map(Closure {}, [ "*** sensi ... "])
/var/www/html/apps/files_sharing/lib/Controller/ShareAPIController.php - line 170: 
OC\Files\Node\Folder->getById(85147)
/var/www/html/apps/files_sharing/lib/Controller/ShareAPIController.php - line 585: 
OCA\Files_Sharing\Controller\ShareAPIController->formatShare(OC\Share20\Share {})
/var/www/html/apps/files_sharing/lib/Controller/ShareAPIController.php - line 679: 
OCA\Files_Sharing\Controller\ShareAPIController->getSharedWithMe(null, true)
/var/www/html/lib/private/AppFramework/Http/Dispatcher.php - line 166: 
OCA\Files_Sharing\Controller\ShareAPIController->getShares("true", "false", "false", null, true)
/var/www/html/lib/private/AppFramework/Http/Dispatcher.php - line 99: 
OC\AppFramework\Http\Dispatcher->executeController(OCA\Files_Sh ... {}, "getShares")
/var/www/html/lib/private/AppFramework/App.php - line 118: 
OC\AppFramework\Http\Dispatcher->dispatch(OCA\Files_Sh ... {}, "getShares")
/var/www/html/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: 
OC\AppFramework\App::main("OCA\\Files_ ... r", "getShares", OC\AppFramew ... {}, { _route: "o ... "})

OC\AppFramework\Routing\RouteActionHandler->__invoke({ _route: "o ... "})
/var/www/html/lib/private/Route/Router.php - line 297: 
call_user_func(OC\AppFramew ... {}, { _route: "o ... "})
/var/www/html/ocs/v1.php - line 82: 
OC\Route\Router->match("/ocsapp/app ... s")
Cased by OCP\Files\NotFoundException: File with id "85147" has not been found.
/var/www/html/apps/files_sharing/lib/SharedStorage.php - line 127: 
OC\Files\View->getPath(85147)
/var/www/html/apps/files_sharing/lib/SharedStorage.php - line 479: 
OCA\Files_Sharing\SharedStorage->init()
/var/www/html/lib/private/Files/Storage/Wrapper/Jail.php - line 218: 
OCA\Files_Sharing\SharedStorage->getWrapperStorage()
/var/www/html/apps/files_sharing/lib/SharedStorage.php - line 198: 
OC\Files\Storage\Wrapper\Jail->getPermissions("")
/var/www/html/lib/private/Files/Storage/Wrapper/Wrapper.php - line 214: 
OCA\Files_Sharing\SharedStorage->getPermissions("")
/var/www/html/apps/files_sharing/lib/Cache.php - line 145: 
OC\Files\Storage\Wrapper\Wrapper->getPermissions("")
/var/www/html/apps/files_sharing/lib/Cache.php - line 113: 
OCA\Files_Sharing\Cache->formatCacheEntry(OC\Files\Cache\CacheEntry {}, "")
/var/www/html/lib/private/Files/Node/Folder.php - line 307: 
OCA\Files_Sharing\Cache->get(85147)

OC\Files\Node\Folder->OC\Files\Node\{closure}("*** sensiti ... *")
/var/www/html/lib/private/Files/Node/Folder.php - line 321: 
array_map(Closure {}, [ "*** sensi ... "])
/var/www/html/apps/files_sharing/lib/Controller/ShareAPIController.php - line 170: 
OC\Files\Node\Folder->getById(85147)
/var/www/html/apps/files_sharing/lib/Controller/ShareAPIController.php - line 585: 
OCA\Files_Sharing\Controller\ShareAPIController->formatShare(OC\Share20\Share {})
/var/www/html/apps/files_sharing/lib/Controller/ShareAPIController.php - line 679: 
OCA\Files_Sharing\Controller\ShareAPIController->getSharedWithMe(null, true)
/var/www/html/lib/private/AppFramework/Http/Dispatcher.php - line 166: 
OCA\Files_Sharing\Controller\ShareAPIController->getShares("true", "false", "false", null, true)
/var/www/html/lib/private/AppFramework/Http/Dispatcher.php - line 99: 
OC\AppFramework\Http\Dispatcher->executeController(OCA\Files_Sh ... {}, "getShares")
/var/www/html/lib/private/AppFramework/App.php - line 118: 
OC\AppFramework\Http\Dispatcher->dispatch(OCA\Files_Sh ... {}, "getShares")
/var/www/html/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: 
OC\AppFramework\App::main("OCA\\Files_ ... r", "getShares", OC\AppFramew ... {}, { _route: "o ... "})

OC\AppFramework\Routing\RouteActionHandler->__invoke({ _route: "o ... "})
/var/www/html/lib/private/Route/Router.php - line 297: 
call_user_func(OC\AppFramew ... {}, { _route: "o ... "})
/var/www/html/ocs/v1.php - line 82: 
OC\Route\Router->match("/ocsapp/app ... s")

```

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>
This user had no issues at all with NC before the upgrade. Problem came up after. I am seeing no other users impacted with this issue, which is odd.

**Comment by Tarry91** [comment]
I do have the same problem for one of our users. I am available to post any information if necessary.

**Comment by jospoortvliet** [comment]
Should be fixed in 14.0.2, test the release candidate - now out. https://help.nextcloud.com/t/help-testing-rc1-version-of-14-0-2-13-0-7-and-12-0-12/38359
